<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Marlene Ganslmeier" />


<title>Template Title</title>

<script src="site_libs/header-attrs-2.11.3/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Mouse_ACT_Seq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/SportyScientist/Mouse_ACT_Seq">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Template Title</h1>
<h4 class="author">Marlene Ganslmeier</h4>
<h4 class="date">25 October, 2021</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-10-25
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>Mouse_ACT_Seq/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20210715code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20210715)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20210715code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20210715)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomSportyScientistMouseACTSeqtree8bd8dd3ce80a4e777a1ebf18827bb4d7be4b29batargetblank8bd8dd3a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/SportyScientist/Mouse_ACT_Seq/tree/8bd8dd3ce80a4e777a1ebf18827bb4d7be4b29ba" target="_blank">8bd8dd3</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomSportyScientistMouseACTSeqtree8bd8dd3ce80a4e777a1ebf18827bb4d7be4b29batargetblank8bd8dd3a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/SportyScientist/Mouse_ACT_Seq/tree/8bd8dd3ce80a4e777a1ebf18827bb4d7be4b29ba" target="_blank">8bd8dd3</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .RData

Untracked files:
    Untracked:  Rnbeads_Best_practice_10_21_files/
    Untracked:  Rplots.pdf
    Untracked:  Untitled/
    Untracked:  analysis/ACT_Seq_K9_broadpeaks.Rmd
    Untracked:  analysis/newpipeline.Rmd
    Untracked:  clustering_all_marks.Rmd
    Untracked:  code/bigwigs.sh
    Untracked:  code/bigwigs_RPKM.sh
    Untracked:  code/matrix_RPKM.sh
    Untracked:  code/matrixgeneration.sh
    Untracked:  data/chmm_mESC.rds
    Untracked:  data/chmm_segments_mm10.rds
    Untracked:  figure/
    Untracked:  matrixcalc.sh
    Untracked:  output/datasummedH3K27ac.rds
    Untracked:  output/datasummedH3K27me3.rds
    Untracked:  output/datasummedH3K36me3.rds
    Untracked:  output/datasummedH3K9me3.rds
    Untracked:  output/datasummed_none.rds
    Untracked:  output/datasummed_norm_BPM.rds
    Untracked:  output/datasummed_norm_CPM.rds
    Untracked:  output/datasummed_norm_MACS2.rds
    Untracked:  output/datasummed_norm_RPGC.rds
    Untracked:  output/datasummed_norm_RPKM.rds
    Untracked:  output/datasummed_norm_cwlab.rds
    Untracked:  output/datasummed_normcomp.rds
    Untracked:  output/sum_full_data.Rds
    Untracked:  output/sum_full_data_predelta.Rds
    Untracked:  peakfiles.csv

Unstaged changes:
    Deleted:    analysis/methylation_domain_analysis.Rmd
    Modified:   analysis/normalization_comparison.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/Diffbind_K9me3.Rmd</code>) and HTML (<code>docs/Diffbind_K9me3.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/SportyScientist/Mouse_ACT_Seq/blob/8bd8dd3ce80a4e777a1ebf18827bb4d7be4b29ba/analysis/Diffbind_K9me3.Rmd" target="_blank">8bd8dd3</a>
</td>
<td>
margan96
</td>
<td>
2021-10-25
</td>
<td>
initial publish
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>##edit name depending on what you want to compute!
NAME &lt;- &quot;ACT_seq_K9me3&quot;
EXT &lt;- &quot;_dba_results.rds&quot;
path &lt;- paste(&quot;/icgc/dkfzlsdf/analysis/C010/ganslm/&quot;, NAME, EXT)


# create sample sheet 

BAM_DIR = c(&quot;/icgc/dkfzlsdf/analysis/C010/cwlab_processing_out2/canepi-srv1/20210309_120_H33_Ilse20790/runs_out&quot;,
 &quot;/icgc/dkfzlsdf/analysis/C010/cwlab_processing_out2/canepi-srv1/20201225_629_HistMut_20245/runs_out&quot;,
 &quot;/icgc/dkfzlsdf/analysis/C010/cwlab_processing_out2/canepi-srv1/20210311_219_H33_Ilse20866/runs_out&quot;,
 &quot;/icgc/dkfzlsdf/analysis/C010/cwlab_processing_out2/canepi-srv1/20201225_122_HistMut_20247/runs_out&quot;,
 &quot;/icgc/dkfzlsdf/analysis/C010/cwlab_processing_out2/canepi-srv1/20201225_032_HistMut_20244/runs_out&quot;,
 &quot;/icgc/dkfzlsdf/analysis/C010/cwlab_processing_out2/canepi-srv1/20210316_239_H33_Ilse20926/runs_out&quot;,
 &quot;/icgc/dkfzlsdf/analysis/C010/cwlab_processing_out2/canepi-srv1/20201225_988_HistMut_20246/runs_out&quot;,
 &quot;/icgc/dkfzlsdf/analysis/C010/cwlab_processing_out2/canepi-srv1/20201225_548_HistMut_20250/runs_out&quot;,
 &quot;/omics/groups/OE0219/internal/ganslm/Analysis/Mouse_ActSeq/ACTseq/cwlab_outputs/20200621_532_ACT-seq_ESC_histone_3_2/runs_out&quot;,
 &quot;/omics/groups/OE0219/internal/ganslm/Analysis/Mouse_ActSeq/ACTseq/cwlab_outputs/20200610_204_ACT-seq_ESC_histone3/&quot;)


PEAK_DIR &lt;- c(&quot;/omics/groups/OE0219/internal/ganslm/Analysis/Mouse_ActSeq/New/20201225_032_HistMut_20244&quot;,
&quot;/omics/groups/OE0219/internal/ganslm/Analysis/Mouse_ActSeq/New/20201225_122_HistMut_20247&quot;,
&quot;/omics/groups/OE0219/internal/ganslm/Analysis/Mouse_ActSeq/New/20201225_548_HistMut_20250&quot;,
&quot;/omics/groups/OE0219/internal/ganslm/Analysis/Mouse_ActSeq/New/20201225_629_HistMut_20245&quot;,
&quot;/omics/groups/OE0219/internal/ganslm/Analysis/Mouse_ActSeq/New/20201225_988_HistMut_20246&quot;,
&quot;/omics/groups/OE0219/internal/ganslm/Analysis/Mouse_ActSeq/New/20210309_120_H33_Ilse20790&quot;,
&quot;/omics/groups/OE0219/internal/ganslm/Analysis/Mouse_ActSeq/New/20210311_219_H33_Ilse20866&quot;,
&quot;/omics/groups/OE0219/internal/ganslm/Analysis/Mouse_ActSeq/New/20210316_239_H33_Ilse20926&quot;)


#for H3K9me3

bamReads &lt;- list.files(path = BAM_DIR, pattern = glob2rx(&quot;*tn5correct.bam&quot;),
                        recursive = T, 
                        full.names = TRUE)
bamReads &lt;- bamReads[grep(&quot;H3K9me3&quot;, bamReads)]


Peaks &lt;- list.files(path = PEAK_DIR,
                    pattern = glob2rx(&quot;*_peaks.xls&quot;),
                    recursive = T, 
                    full.names = TRUE)
Peaks &lt;- Peaks[grep(&quot;H3K9me3&quot;, Peaks)]
 

SampleID &lt;- sapply(strsplit(split=&quot;/&quot;,bamReads), &quot;[&quot;, 10)

Condition &lt;-  sapply(strsplit(split=&quot;_&quot;, SampleID), &quot;[&quot;, 1)

Sequencingrun &lt;- sapply(strsplit(split=&quot;/&quot;,bamReads), &quot;[&quot;, 8)

samples_K9 &lt;- data.frame(SampleID, Tissue = &quot;Cell_line&quot;, Factor = &quot;H3K9me3&quot;, Condition, bamReads, Peaks, Sequencingrun, &quot;PeakCaller&quot; =  &quot;macs&quot;)

samples_K9</code></pre>
<pre><code>          SampleID    Tissue  Factor Condition
1  KH2G34L_H3K9me3 Cell_line H3K9me3   KH2G34L
2  KH2G34R_H3K9me3 Cell_line H3K9me3   KH2G34R
3  KH2G34W_H3K9me3 Cell_line H3K9me3   KH2G34W
4    KH2wt_H3K9me3 Cell_line H3K9me3     KH2wt
5  KH2G34L_H3K9me3 Cell_line H3K9me3   KH2G34L
6  KH2G34R_H3K9me3 Cell_line H3K9me3   KH2G34R
7    KH2wt_H3K9me3 Cell_line H3K9me3     KH2wt
8  KH2G34L_H3K9me3 Cell_line H3K9me3   KH2G34L
9  KH2G34R_H3K9me3 Cell_line H3K9me3   KH2G34R
10 KH2G34W_H3K9me3 Cell_line H3K9me3   KH2G34W
11   KH2wt_H3K9me3 Cell_line H3K9me3     KH2wt
                                                                                                                                                 bamReads
1  /icgc/dkfzlsdf/analysis/C010/cwlab_processing_out2/canepi-srv1/20201225_548_HistMut_20250/runs_out/KH2G34L_H3K9me3/KH2G34L_H3K9me3_filt_tn5correct.bam
2  /icgc/dkfzlsdf/analysis/C010/cwlab_processing_out2/canepi-srv1/20201225_548_HistMut_20250/runs_out/KH2G34R_H3K9me3/KH2G34R_H3K9me3_filt_tn5correct.bam
3  /icgc/dkfzlsdf/analysis/C010/cwlab_processing_out2/canepi-srv1/20201225_548_HistMut_20250/runs_out/KH2G34W_H3K9me3/KH2G34W_H3K9me3_filt_tn5correct.bam
4      /icgc/dkfzlsdf/analysis/C010/cwlab_processing_out2/canepi-srv1/20201225_548_HistMut_20250/runs_out/KH2wt_H3K9me3/KH2wt_H3K9me3_filt_tn5correct.bam
5  /icgc/dkfzlsdf/analysis/C010/cwlab_processing_out2/canepi-srv1/20201225_629_HistMut_20245/runs_out/KH2G34L_H3K9me3/KH2G34L_H3K9me3_filt_tn5correct.bam
6  /icgc/dkfzlsdf/analysis/C010/cwlab_processing_out2/canepi-srv1/20201225_629_HistMut_20245/runs_out/KH2G34R_H3K9me3/KH2G34R_H3K9me3_filt_tn5correct.bam
7      /icgc/dkfzlsdf/analysis/C010/cwlab_processing_out2/canepi-srv1/20201225_629_HistMut_20245/runs_out/KH2wt_H3K9me3/KH2wt_H3K9me3_filt_tn5correct.bam
8  /icgc/dkfzlsdf/analysis/C010/cwlab_processing_out2/canepi-srv1/20210309_120_H33_Ilse20790/runs_out/KH2G34L_H3K9me3/KH2G34L_H3K9me3_filt_tn5correct.bam
9  /icgc/dkfzlsdf/analysis/C010/cwlab_processing_out2/canepi-srv1/20210309_120_H33_Ilse20790/runs_out/KH2G34R_H3K9me3/KH2G34R_H3K9me3_filt_tn5correct.bam
10 /icgc/dkfzlsdf/analysis/C010/cwlab_processing_out2/canepi-srv1/20210309_120_H33_Ilse20790/runs_out/KH2G34W_H3K9me3/KH2G34W_H3K9me3_filt_tn5correct.bam
11     /icgc/dkfzlsdf/analysis/C010/cwlab_processing_out2/canepi-srv1/20210309_120_H33_Ilse20790/runs_out/KH2wt_H3K9me3/KH2wt_H3K9me3_filt_tn5correct.bam
                                                                                                                                 Peaks
1  /omics/groups/OE0219/internal/ganslm/Analysis/Mouse_ActSeq/New/20201225_548_HistMut_20250/KH2G34L_H3K9me3_filt_tn5correct_peaks.xls
2  /omics/groups/OE0219/internal/ganslm/Analysis/Mouse_ActSeq/New/20201225_548_HistMut_20250/KH2G34R_H3K9me3_filt_tn5correct_peaks.xls
3  /omics/groups/OE0219/internal/ganslm/Analysis/Mouse_ActSeq/New/20201225_548_HistMut_20250/KH2G34W_H3K9me3_filt_tn5correct_peaks.xls
4    /omics/groups/OE0219/internal/ganslm/Analysis/Mouse_ActSeq/New/20201225_548_HistMut_20250/KH2wt_H3K9me3_filt_tn5correct_peaks.xls
5  /omics/groups/OE0219/internal/ganslm/Analysis/Mouse_ActSeq/New/20201225_629_HistMut_20245/KH2G34L_H3K9me3_filt_tn5correct_peaks.xls
6  /omics/groups/OE0219/internal/ganslm/Analysis/Mouse_ActSeq/New/20201225_629_HistMut_20245/KH2G34R_H3K9me3_filt_tn5correct_peaks.xls
7    /omics/groups/OE0219/internal/ganslm/Analysis/Mouse_ActSeq/New/20201225_629_HistMut_20245/KH2wt_H3K9me3_filt_tn5correct_peaks.xls
8  /omics/groups/OE0219/internal/ganslm/Analysis/Mouse_ActSeq/New/20210309_120_H33_Ilse20790/KH2G34L_H3K9me3_filt_tn5correct_peaks.xls
9  /omics/groups/OE0219/internal/ganslm/Analysis/Mouse_ActSeq/New/20210309_120_H33_Ilse20790/KH2G34R_H3K9me3_filt_tn5correct_peaks.xls
10 /omics/groups/OE0219/internal/ganslm/Analysis/Mouse_ActSeq/New/20210309_120_H33_Ilse20790/KH2G34W_H3K9me3_filt_tn5correct_peaks.xls
11   /omics/groups/OE0219/internal/ganslm/Analysis/Mouse_ActSeq/New/20210309_120_H33_Ilse20790/KH2wt_H3K9me3_filt_tn5correct_peaks.xls
                Sequencingrun PeakCaller
1  20201225_548_HistMut_20250       macs
2  20201225_548_HistMut_20250       macs
3  20201225_548_HistMut_20250       macs
4  20201225_548_HistMut_20250       macs
5  20201225_629_HistMut_20245       macs
6  20201225_629_HistMut_20245       macs
7  20201225_629_HistMut_20245       macs
8  20210309_120_H33_Ilse20790       macs
9  20210309_120_H33_Ilse20790       macs
10 20210309_120_H33_Ilse20790       macs
11 20210309_120_H33_Ilse20790       macs</code></pre>
<pre class="r"><code>peaks &lt;- dba(sampleSheet = samples_K9)
peaks &lt;- dba.blacklist(peaks, blacklist = DBA_BLACKLIST_HG19, greylist=FALSE)
peaks &lt;- dba.count(peaks)
peaks</code></pre>
<pre><code>11 Samples, 131698 sites in matrix:
                ID    Tissue  Factor Condition    Reads FRiP
1  KH2G34L_H3K9me3 Cell_line H3K9me3   KH2G34L  8807181 0.14
2  KH2G34R_H3K9me3 Cell_line H3K9me3   KH2G34R  9879046 0.26
3  KH2G34W_H3K9me3 Cell_line H3K9me3   KH2G34W 11120888 0.22
4    KH2wt_H3K9me3 Cell_line H3K9me3     KH2wt  8214506 0.14
5  KH2G34L_H3K9me3 Cell_line H3K9me3   KH2G34L 13816092 0.19
6  KH2G34R_H3K9me3 Cell_line H3K9me3   KH2G34R  8779759 0.15
7    KH2wt_H3K9me3 Cell_line H3K9me3     KH2wt 15837093 0.10
8  KH2G34L_H3K9me3 Cell_line H3K9me3   KH2G34L 13210888 0.08
9  KH2G34R_H3K9me3 Cell_line H3K9me3   KH2G34R  7615363 0.09
10 KH2G34W_H3K9me3 Cell_line H3K9me3   KH2G34W  9254629 0.19
11   KH2wt_H3K9me3 Cell_line H3K9me3     KH2wt 10927494 0.24</code></pre>
<pre class="r"><code>plot(peaks)</code></pre>
<p><img src="figure/Diffbind_K9me3.Rmd/unnamed-chunk-2-1.png" width="1440" style="display: block; margin: auto;" /></p>
<pre class="r"><code>peaks &lt;- dba.normalize(peaks)</code></pre>
<div id="section" class="section level1 tabset tabset-fade tabset-pills">
<h1 class="tabset tabset-fade tabset-pills"></h1>
<pre class="r"><code>## PCA
dba.plotPCA(peaks, label= DBA_CONDITION)</code></pre>
<p><img src="figure/Diffbind_K9me3.Rmd/unnamed-chunk-3-1.png" width="1440" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## MA
dba.plotMA(peaks, bNormalized=FALSE,
           contrast=list(wt=peaks$masks$KH2wt,
                         G34W=peaks$masks$KH2G34W))</code></pre>
<p><img src="figure/Diffbind_K9me3.Rmd/unnamed-chunk-3-2.png" width="1440" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#this models the data and sets wt as &quot;base&quot;
contrast &lt;- dba.contrast(peaks, reorderMeta=list(Condition=&quot;KH2wt&quot;))

#differential analyis with standard FDR of &lt;0.05
diffbind &lt;- dba.analyze(contrast)
dba.show(diffbind, bContrasts = T)</code></pre>
<pre><code>     Factor   Group Samples  Group2 Samples2 DB.DESeq2
1 Condition KH2G34L       3 KH2G34R        3         0
2 Condition KH2G34L       3   KH2wt        3         0
3 Condition KH2G34R       3   KH2wt        3         0</code></pre>
<pre class="r"><code>plot(diffbind, contrast = 1)</code></pre>
<pre><code>Error: Unable to plot -- no sites within threshold</code></pre>
<pre class="r"><code>db_sites &lt;- dba.report(diffbind)
#saveRDS(db_sites, file = &quot;/icgc/dkfzlsdf/analysis/C010/ganslm/workflowr/ACT_Seq2/output/db_sites.rds&quot;)
#db_sites &lt;- readRDS(&quot;/icgc/dkfzlsdf/analysis/C010/ganslm/workflowr/ACT_Seq2/output/db_sites.rds&quot;)

#plots them as Venn 
## Venn
dba.plotVenn(diffbind, contrast=1, bDB=TRUE, bGain=TRUE, bLoss=TRUE, bAll=FALSE)</code></pre>
<pre><code>Error: No valid contrasts/methods specified.</code></pre>
<pre class="r"><code>gain &lt;- db_sites[db_sites$Fold &gt; 0]
loss &lt;- db_sites[db_sites$Fold &lt; 0]

# Create GRanges object with annotations from TxDb database
annoData &lt;- toGRanges(TxDb.Hsapiens.UCSC.hg19.knownGene, feature=&quot;gene&quot;)


# Load mart
ensembl &lt;- useEnsembl(biomart = &quot;genes&quot;,
                     dataset = &quot;hsapiens_gene_ensembl&quot;)</code></pre>
<p>First look at the peaks that were gained</p>
<pre class="r"><code>genomicElementDistribution(gain, 
                           TxDb = TxDb.Hsapiens.UCSC.hg19.knownGene,
                           promoterRegion=c(upstream=2000, downstream=500),
                           geneDownstream=c(upstream=0, downstream=2000),
                           promoterLevel=list(
                         # from 5&#39; -&gt; 3&#39;, fixed precedence 3&#39; -&gt; 5&#39;
                             breaks = c(-2000, -1000, -500, 0, 500),
                             labels = c(&quot;upstream 1-2Kb&quot;, &quot;upstream 0.5-1Kb&quot;, 
                                        &quot;upstream &lt;500b&quot;, &quot;TSS - 500b&quot;),
                             colors = c(&quot;#FFE5CC&quot;, &quot;#FFCA99&quot;, 
                                        &quot;#FFAD65&quot;, &quot;#FF8E32&quot;)))</code></pre>
<pre><code>Error in genomicElementDistribution(gain, TxDb = TxDb.Hsapiens.UCSC.hg19.knownGene, : peaks must be an object of GRanges or GRangesList</code></pre>
<pre class="r"><code># Annotate granges with the nearest TSS
annot &lt;- annotatePeakInBatch(gain, 
                               AnnotationData=annoData, 
                               featureType = &quot;TSS&quot;,
                               output=&quot;nearestLocation&quot;,
                               PeakLocForDistance = &quot;start&quot;)</code></pre>
<pre><code>Error in annotatePeakInBatch(gain, AnnotationData = annoData, featureType = &quot;TSS&quot;, : No valid myPeakList passed in. It needs to be GRanges object</code></pre>
<pre class="r"><code>annot &lt;- addGeneIDs(annot, mart = ensembl, feature_id_type = &quot;ensembl_gene_id&quot;,
                      IDs2Add = c(&quot;entrezgene_id&quot;, &quot;description&quot;))</code></pre>
<pre><code>Error in is(annotatedPeak, &quot;GRanges&quot;): object &#39;annot&#39; not found</code></pre>
<pre class="r"><code>annot.copy &lt;- annot</code></pre>
<pre><code>Error in eval(expr, envir, enclos): object &#39;annot&#39; not found</code></pre>
<pre class="r"><code>annot.copy$score &lt;- 1</code></pre>
<pre><code>Error in annot.copy$score &lt;- 1: object &#39;annot.copy&#39; not found</code></pre>
<pre class="r"><code>binOverFeature(annot, annot.copy, annotationData=annoData,
               radius=5000, nbins=10, FUN=c(sum, length),
               ylab=c(&quot;score&quot;, &quot;count&quot;), 
               main=c(&quot;Distribution of aggregated peak scores around TSS&quot;, 
                      &quot;Distribution of aggregated peak numbers around TSS&quot;))</code></pre>
<pre><code>Error in binOverFeature(annot, annot.copy, annotationData = annoData, : object &#39;annot&#39; not found</code></pre>
<pre class="r"><code>if(require(TxDb.Hsapiens.UCSC.hg19.knownGene)){
    aCR&lt;-assignChromosomeRegion(annot, nucleotideLevel=FALSE, 
                           precedence=c(&quot;Promoters&quot;, &quot;immediateDownstream&quot;, 
                                         &quot;fiveUTRs&quot;, &quot;threeUTRs&quot;, 
                                         &quot;Exons&quot;, &quot;Introns&quot;), 
                           TxDb=TxDb.Hsapiens.UCSC.hg19.knownGene)
    barplot(aCR$percentage)
}</code></pre>
<pre><code>Error in assignChromosomeRegion(annot, nucleotideLevel = FALSE, precedence = c(&quot;Promoters&quot;, : object &#39;annot&#39; not found</code></pre>
<pre class="r"><code>pie1(table(annot$insideFeature))</code></pre>
<pre><code>Error in eval(quote(list(...)), env): object &#39;annot&#39; not found</code></pre>
<pre class="r"><code>over &lt;- getEnrichedGO(annot, orgAnn=&quot;org.Hs.eg.db&quot;, condense=TRUE, feature_id_type =&quot;entrez_id&quot;)</code></pre>
<pre><code>Error in is(annotatedPeak, &quot;GRangesList&quot;): object &#39;annot&#39; not found</code></pre>
<pre class="r"><code>enrichmentPlot(over)</code></pre>
<pre><code>Error in is.data.frame(res): object &#39;over&#39; not found</code></pre>
<pre class="r"><code>over</code></pre>
<pre><code>Error in eval(expr, envir, enclos): object &#39;over&#39; not found</code></pre>
<pre class="r"><code>path &lt;- getEnrichedPATH(annot, &quot;org.Hs.eg.db&quot;, &quot;reactome.db&quot;, maxP=.05, feature_id_type =&quot;entrez_id&quot;)</code></pre>
<pre><code>Error in getEnrichedPATH(annot, &quot;org.Hs.eg.db&quot;, &quot;reactome.db&quot;, maxP = 0.05, : object &#39;annot&#39; not found</code></pre>
<pre class="r"><code>path &lt;- separate(path, path.term, sep = &quot;:&quot;, into = c(NA, &quot;path.term&quot;))</code></pre>
<pre><code>Error in separate(path, path.term, sep = &quot;:&quot;, into = c(NA, &quot;path.term&quot;)): could not find function &quot;separate&quot;</code></pre>
<pre class="r"><code>path</code></pre>
<pre><code>[1] &quot;/icgc/dkfzlsdf/analysis/C010/ganslm/ ACT_seq_K9me3 _dba_results.rds&quot;</code></pre>
<pre class="r"><code>enrichmentPlot(path)</code></pre>
<pre><code>Error: $ operator is invalid for atomic vectors</code></pre>
<p>gain of peaks –&gt; silencing –&gt; peaks gained whee beta-catenin complex is deactivated –&gt; beta catenin is upregulated –&gt; cancer</p>
<pre class="r"><code>genomicElementDistribution(loss, 
                           TxDb = TxDb.Hsapiens.UCSC.hg19.knownGene,
                           promoterRegion=c(upstream=2000, downstream=500),
                           geneDownstream=c(upstream=0, downstream=2000),
                           promoterLevel=list(
                         # from 5&#39; -&gt; 3&#39;, fixed precedence 3&#39; -&gt; 5&#39;
                             breaks = c(-2000, -1000, -500, 0, 500),
                             labels = c(&quot;upstream 1-2Kb&quot;, &quot;upstream 0.5-1Kb&quot;, 
                                        &quot;upstream &lt;500b&quot;, &quot;TSS - 500b&quot;),
                             colors = c(&quot;#FFE5CC&quot;, &quot;#FFCA99&quot;, 
                                        &quot;#FFAD65&quot;, &quot;#FF8E32&quot;)))</code></pre>
<pre><code>Error in genomicElementDistribution(loss, TxDb = TxDb.Hsapiens.UCSC.hg19.knownGene, : peaks must be an object of GRanges or GRangesList</code></pre>
<pre class="r"><code># Annotate granges with the nearest TSS
annot &lt;- annotatePeakInBatch(loss, 
                               AnnotationData=annoData, 
                               featureType = &quot;TSS&quot;,
                               output=&quot;nearestLocation&quot;,
                               PeakLocForDistance = &quot;start&quot;)</code></pre>
<pre><code>Error in annotatePeakInBatch(loss, AnnotationData = annoData, featureType = &quot;TSS&quot;, : No valid myPeakList passed in. It needs to be GRanges object</code></pre>
<pre class="r"><code>annot &lt;- addGeneIDs(annot, mart = ensembl, feature_id_type = &quot;ensembl_gene_id&quot;,
                      IDs2Add = c(&quot;entrezgene_id&quot;, &quot;description&quot;))</code></pre>
<pre><code>Error in is(annotatedPeak, &quot;GRanges&quot;): object &#39;annot&#39; not found</code></pre>
<pre class="r"><code>annot.copy &lt;- annot</code></pre>
<pre><code>Error in eval(expr, envir, enclos): object &#39;annot&#39; not found</code></pre>
<pre class="r"><code>annot.copy$score &lt;- 1</code></pre>
<pre><code>Error in annot.copy$score &lt;- 1: object &#39;annot.copy&#39; not found</code></pre>
<pre class="r"><code>binOverFeature(annot, annot.copy, annotationData=annoData,
               radius=5000, nbins=10, FUN=c(sum, length),
               ylab=c(&quot;score&quot;, &quot;count&quot;), 
               main=c(&quot;Distribution of aggregated peak scores around TSS&quot;, 
                      &quot;Distribution of aggregated peak numbers around TSS&quot;))</code></pre>
<pre><code>Error in binOverFeature(annot, annot.copy, annotationData = annoData, : object &#39;annot&#39; not found</code></pre>
<pre class="r"><code>if(require(TxDb.Hsapiens.UCSC.hg19.knownGene)){
    aCR&lt;-assignChromosomeRegion(annot, nucleotideLevel=FALSE, 
                           precedence=c(&quot;Promoters&quot;, &quot;immediateDownstream&quot;, 
                                         &quot;fiveUTRs&quot;, &quot;threeUTRs&quot;, 
                                         &quot;Exons&quot;, &quot;Introns&quot;), 
                           TxDb=TxDb.Hsapiens.UCSC.hg19.knownGene)
    barplot(aCR$percentage)
}</code></pre>
<pre><code>Error in assignChromosomeRegion(annot, nucleotideLevel = FALSE, precedence = c(&quot;Promoters&quot;, : object &#39;annot&#39; not found</code></pre>
<pre class="r"><code>pie1(table(annot$insideFeature))</code></pre>
<pre><code>Error in eval(quote(list(...)), env): object &#39;annot&#39; not found</code></pre>
<pre class="r"><code>over &lt;- getEnrichedGO(annot, orgAnn=&quot;org.Hs.eg.db&quot;, condense=TRUE, feature_id_type =&quot;entrez_id&quot;)</code></pre>
<pre><code>Error in is(annotatedPeak, &quot;GRangesList&quot;): object &#39;annot&#39; not found</code></pre>
<pre class="r"><code>enrichmentPlot(over)</code></pre>
<pre><code>Error in is.data.frame(res): object &#39;over&#39; not found</code></pre>
<pre class="r"><code>path &lt;- getEnrichedPATH(annot, &quot;org.Hs.eg.db&quot;, &quot;reactome.db&quot;, maxP=.05, feature_id_type =&quot;entrez_id&quot;)</code></pre>
<pre><code>Error in getEnrichedPATH(annot, &quot;org.Hs.eg.db&quot;, &quot;reactome.db&quot;, maxP = 0.05, : object &#39;annot&#39; not found</code></pre>
<pre class="r"><code>path &lt;- separate(path, path.term, sep = &quot;:&quot;, into = c(NA, &quot;path.term&quot;))</code></pre>
<pre><code>Error in separate(path, path.term, sep = &quot;:&quot;, into = c(NA, &quot;path.term&quot;)): could not find function &quot;separate&quot;</code></pre>
<pre class="r"><code>enrichmentPlot(path)</code></pre>
<pre><code>Error: $ operator is invalid for atomic vectors</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.1 (2021-08-10)
Platform: x86_64-conda-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /omics/groups/OE0219/internal/ganslm/miniconda3/envs/R4.1/lib/libopenblasp-r0.3.17.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] stringr_1.4.0                          
 [2] reactome.db_1.76.0                     
 [3] ChIPpeakAnno_3.26.4                    
 [4] Repitools_1.38.0                       
 [5] org.Hs.eg.db_3.13.0                    
 [6] gprofiler2_0.2.1                       
 [7] TxDb.Hsapiens.UCSC.hg19.knownGene_3.2.2
 [8] GenomicFeatures_1.44.2                 
 [9] AnnotationDbi_1.54.1                   
[10] biomaRt_2.48.3                         
[11] csaw_1.26.0                            
[12] dplyr_1.0.7                            
[13] BiocParallel_1.26.2                    
[14] DiffBind_3.2.7                         
[15] SummarizedExperiment_1.22.0            
[16] Biobase_2.52.0                         
[17] MatrixGenerics_1.4.3                   
[18] matrixStats_0.61.0                     
[19] GenomicRanges_1.44.0                   
[20] GenomeInfoDb_1.28.4                    
[21] IRanges_2.26.0                         
[22] S4Vectors_0.30.2                       
[23] BiocGenerics_0.38.0                    
[24] workflowr_1.6.2                        

loaded via a namespace (and not attached):
  [1] utf8_1.2.2               tidyselect_1.1.1         RSQLite_2.2.8           
  [4] htmlwidgets_1.5.4        grid_4.1.1               munsell_0.5.0           
  [7] base64url_1.4            preprocessCore_1.54.0    systemPipeR_1.26.3      
 [10] withr_2.4.2              colorspace_2.0-2         Category_2.58.0         
 [13] filelock_1.0.2           highr_0.9                knitr_1.36              
 [16] git2r_0.28.0             bbmle_1.0.24             GenomeInfoDbData_1.2.6  
 [19] mixsqp_0.3-43            hwriter_1.3.2            bit64_4.0.5             
 [22] pheatmap_1.0.12          rprojroot_2.0.2          coda_0.19-4             
 [25] batchtools_0.9.15        vctrs_0.3.8              generics_0.1.0          
 [28] lambda.r_1.2.4           xfun_0.27                BiocFileCache_2.0.0     
 [31] regioneR_1.24.0          R6_2.5.1                 apeglm_1.14.0           
 [34] invgamma_1.1             locfit_1.5-9.4           rsvg_2.1.2              
 [37] AnnotationFilter_1.16.0  bitops_1.0-7             cachem_1.0.6            
 [40] DelayedArray_0.18.0      assertthat_0.2.1         Ringo_1.56.0            
 [43] promises_1.2.0.1         BiocIO_1.2.0             scales_1.1.1            
 [46] gtable_0.3.0             affy_1.70.0              ensembldb_2.16.4        
 [49] rlang_0.4.12             genefilter_1.74.0        splines_4.1.1           
 [52] rtracklayer_1.52.1       lazyeval_0.2.2           brew_1.0-6              
 [55] checkmate_2.0.0          BiocManager_1.30.16      yaml_2.2.1              
 [58] backports_1.2.1          httpuv_1.6.3             RBGL_1.68.0             
 [61] tools_4.1.1              ggplot2_3.3.5            affyio_1.62.0           
 [64] ellipsis_0.3.2           gplots_3.1.1             Rsolnp_1.16             
 [67] jquerylib_0.1.4          RColorBrewer_1.1-2       DNAcopy_1.66.0          
 [70] Rcpp_1.0.7               plyr_1.8.6               progress_1.2.2          
 [73] zlibbioc_1.38.0          purrr_0.3.4              RCurl_1.98-1.5          
 [76] prettyunits_1.1.1        ashr_2.2-47              cluster_2.1.2           
 [79] ggrepel_0.9.1            fs_1.5.0                 magrittr_2.0.1          
 [82] futile.options_1.0.1     data.table_1.14.2        truncnorm_1.0-8         
 [85] mvtnorm_1.1-3            SQUAREM_2021.1           whisker_0.4             
 [88] amap_0.8-18              ProtGenerics_1.24.0      hms_1.1.1               
 [91] evaluate_0.14            xtable_1.8-4             XML_3.99-0.8            
 [94] VennDiagram_1.6.20       emdbook_1.3.12           jpeg_0.1-9              
 [97] compiler_4.1.1           bdsmatrix_1.3-4          tibble_3.1.5            
[100] KernSmooth_2.23-20       V8_3.4.2                 crayon_1.4.1            
[103] htmltools_0.5.2          GOstats_2.58.0           later_1.3.0             
[106] geneplotter_1.70.0       tidyr_1.1.4              DBI_1.1.1               
[109] formatR_1.11             dbplyr_2.1.1             MASS_7.3-54             
[112] rappdirs_0.3.3           ShortRead_1.50.0         Matrix_1.3-4            
[115] vsn_3.60.0               metapod_1.0.0            pkgconfig_2.0.3         
[118] GenomicAlignments_1.28.0 numDeriv_2016.8-1.1      plotly_4.10.0           
[121] InteractionSet_1.20.0    xml2_1.3.2               annotate_1.70.0         
[124] bslib_0.3.1              multtest_2.48.0          XVector_0.32.0          
[127] AnnotationForge_1.34.1   VariantAnnotation_1.38.0 digest_0.6.28           
[130] graph_1.70.0             Biostrings_2.60.2        rmarkdown_2.11.3        
[133] edgeR_3.34.1             GSEABase_1.54.0          restfulr_0.0.13         
[136] GreyListChIP_1.24.0      curl_4.3.2               Rsamtools_2.8.0         
[139] gtools_3.9.2             rjson_0.2.20             lifecycle_1.0.1         
[142] jsonlite_1.7.2           futile.logger_1.4.3      viridisLite_0.4.0       
[145] limma_3.48.3             BSgenome_1.60.0          fansi_0.5.0             
[148] pillar_1.6.4             lattice_0.20-45          KEGGREST_1.32.0         
[151] fastmap_1.1.0            httr_1.4.2               survival_3.2-13         
[154] GO.db_3.13.0             glue_1.4.2               png_0.1-7               
[157] bit_4.0.4                Rgraphviz_2.36.0         stringi_1.7.5           
[160] sass_0.4.0               blob_1.2.2               DESeq2_1.32.0           
[163] latticeExtra_0.6-29      caTools_1.18.2           memoise_2.0.0           
[166] gsmoothr_0.1.7           DOT_0.1                  irlba_2.3.3             </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
